src/neoxp/Extensions/Extensions.cs

```diff
@@ -57,6 +58,28 @@ public static void WriteWarning(string value)
             Console.WriteLine($"\x1b[33mWarning: {value}\x1b[0m");
         }
 
+        public static bool TryGetUtf8String(this ReadOnlySpan<byte> data, out string text)
+        {
+            try
+            {
+                text = Encoding.UTF8.GetString(data);
+                byte[] reencodedBytes = Encoding.UTF8.GetBytes(text);
+                for (int i = 0; i < data.Length; i++)
+                {
+                    if (data[i] != reencodedBytes[i])
```

Comment:
Could check for  `char.IsAscii()`

Reply:
I'm sorry I don't get it. If the data is encoded utf-8 string, checking `IsAscii` is not enough.

Reply:
You allow strings to be displayed that are not human readable.

Reply:
If utf-8 convert successfully, it is already readable without extra checking.

Reply:
ok thought you were using ASCII for some reason... 😂